yt_georaster/data_structures.py

```diff
@@ -324,9 +326,71 @@ def _parse_parameter_file(self):
                 v = f.meta[key]
                 self.parameters[key] = v
             self.parameters["res"] = f.res
-            self.parameters["src"] = f.crs
         self.current_time = 0
 
+        # overwrite crs if one is provided by user
+        if not (self.crs is None):
+            # make sure user provided CRS is valid CRS object
+            if not isinstance(self.crs, CRS):
+                if isinstance(self.crs, int):
+                    # assume epsg number
+                    self.crs = CRS.from_epsg(self.crs)
+                elif isinstance(self.crs, dict):
+                    self.crs = CRS.from_dict(**self.crs)
+                else:
+                    self.crs = CRS.from_string(self.crs)
+
+            # get reprojected transform
+            left_edge = self.parameters["transform"] * (0, 0)
+            right_edge = self.parameters["transform"] * (
+                self.parameters["width"],
+                self.parameters["height"]
+            )
+            transform, width, height = warp.calculate_default_transform(
+                self.parameters["crs"],
+                self.crs,
+                self.parameters["width"],
+                self.parameters["height"],
+                left=left_edge[0],
+                bottom=left_edge[1],
+                right=right_edge[0],
+                top=right_edge[1],
+                # resolution=self.parameters["transform"][0]
+                dst_width=self.parameters["width"],
+                dst_height=self.parameters["height"]
+            )  # current solution can create rectangular pixels
```

Comment:
The above function can be used to get an ideal transform which tries to make sure we get square pixels. As the reprojection changes the shape of the window this means the number of pixels in x and y are changed (and by different amounts). For some reason this then through an error later in the code which I couldn't resolve.

This solution forces the reprojected image to have the same width and height (number of pixels in x and y) but then gives us rectangular pixels! I am a bit concerned that we are already doing something like this without realising in our current solution when we reproject the window already.